src/ff7/widescreen.cpp

```diff
@@ -24,12 +24,28 @@
 #include "widescreen.h"
 #include "../patch.h"
 #include "../ff7.h"
+#include "../cfg.h"
+#include "../renderer.h"
 #include "cmath"
 
 int viewport_width_plus_x_widescreen_fix = 750;
 int swirl_framebuffer_offset_x_widescreen_fix = 106;
 int swirl_framebuffer_offset_y_widescreen_fix = 64;
 
+Widescreen widescreen;
+
+// This function should be called at each frame after drawing backgrounds and 3d models
+void ff7_field_draw_gray_quads_sub_644E90() {
+    ff7_externals.field_draw_gray_quads_644E90();
+
+    if(gl_defer_zoom()) return;
+
+    if(aspect_ratio == AR_WIDESCREEN && widescreen.getMode() == WM_ZOOM)
+        newRenderer.zoomBackendFrameBuffer();
+
+    ffnx_trace("FIELD: draw gray quads\n");
```

Comment:
I left this trace here, but it was meant to be deleted

Reply:
Deleted the trace, thanks.